packages/code/src/renderingPlugin.tsx

```diff
@@ -151,46 +175,131 @@ export function LineTableRow({
   );
 }
 
+interface FlatTokenObject {
+  kind: string;
+  children: Array<string>;
+}
+
+// Check if object is a TokenObject which has an array with a single string element within it.
+function isFlattenedTokenObject(obj: TokenObject): obj is FlatTokenObject {
+  // default to an empty object in the off-chance "obj" is null or undefined.
+  const { children } = obj ?? {};
+
+  if (isArray(children) && children.length === 1 && isString(children[0])) {
+    return true;
+  }
+
+  return false;
+}
+
+// If an array of tokens contains an object with more than one children, this function will flatten that tree recursively.
+function flattenNestedTree(
+  children: Array<string | TokenObject>,
```

Comment:
we could make this `children: TokenObject['children']`, to avoid some redundancy and make it clear where we expect this to come from

Reply:
That's fair. Will do